antsibull/docsite_docs.py

```diff
@@ -0,0 +1,234 @@
+    errors.extend(toctree_errors)
+    if not toctree:
+        errors.append(
+            f'Section #{section_index} has no content')
+        return None, errors
+    return Section(yaml_section['title'], toctree), errors
+
+
+def load_docsite_index(index_path: str) -> t.Tuple[t.List[Section], t.List[str]]:
+    '''Load a docsite extra-docs.yml file.
+
+    :arg index_path: Path to extra-docs.yml (does not need to exist).
+    :arg report_not_existing: Whether to report when extra-docs.yml does not exist.
+    :returns: A tuple consisting of a list of sections and a list of error messages.
+    :raises: DocsiteIndexError if extra-docs.yml does not exist
+    '''
+    sections: t.List[Section] = []
+    errors: t.List[str] = []
+
+    if not os.path.isfile(index_path):
+        raise DocsiteIndexError('extra-docs.yml does not exist')
+
+    try:
+        index = load_yaml_file(index_path)
+        if index.get('sections'):
+            for section_index, yaml_section in enumerate(index['sections']):
+                if not isinstance(yaml_section, dict):
+                    errors.append(f'Section #{section_index} must be a mapping')
+                    continue
+                section, section_errors = load_section(yaml_section, section_index)
+                if section is not None:
+                    sections.append(section)
+                errors.extend(section_errors)
+    except Exception as exc:
+        errors.append(str(exc))
+
+    return sections, errors
+
+
+async def load_collection_docsite(collection_name: str,
+                                  collection_metadata: AnsibleCollectionMetadata,
+                                  path_prefix: str = 'docsite/'
+                                  ) -> CollectionDocsiteInfoT:
+    '''Given a collection name and collection metadata, load docsite data.
+
+    :arg collection_name: Dotted collection name.
+    :arg collection_metadata: Collection metadata object.
+    :arg path_prefix: Prefix to add to relative paths, and toctree entries.
+    :returns: A tuple consisting of a list of sections and a list of RST documents as tuples
+              (relative path in docsite, content).
+    '''
+    flog = mlog.fields(func='load_collection_docsite')
+    flog.debug('Enter')
+
+    index_path = os.path.join(collection_metadata.path, 'docs', 'docsite', 'extra-docs.yml')
+    try:
+        sections, _ = load_docsite_index(index_path)
+    except DocsiteIndexError:
+        sections = []
+
+    for section in sections:
+        for i, toctree in enumerate(section.toctree):
+            section.toctree[i] = path_prefix + toctree
+    documents = []
+    for doc in find_docsite_docs(collection_metadata.path):
+        try:
+            # Load content
+            async with aiofiles.open(doc[0], 'r', encoding='utf-8') as f:
+                content = await f.read()
+
+            # Lint content
+            _, errors = lint_required_conditions(content, collection_name)
```

Comment:
I generally try to avoid using ` bare `_` as an identifier since if we ever i18n'ize this program we'll want to use that for gettext.  In ansible-core they use `dummy` for variables that should be ignored.  Here's pylint's regex:
```
dummy-variables-rgx=_+$|(_[a-zA-Z0-9_]*[a-zA-Z0-9]+?$)|dummy|^ignored_|^unused_
```

Reply:
I've renamed all occurences of `_` to `dummy`.